cmd/oras/internal/errors/errors.go

```diff
@@ -173,3 +173,20 @@ func CheckMutuallyExclusiveFlags(fs *pflag.FlagSet, exclusiveFlagSet ...string)
 	}
 	return nil
 }
+
+// CheckRequiredTogetherFlags checks if any flags required together are all used,
+// returns an error when detecting any flags not used while other flags have been used.
+func CheckRequiredTogetherFlags(fs *pflag.FlagSet, requiredTogetherFlags ...string) error {
+	var unchangedFlags []string
+	for _, flagName := range requiredTogetherFlags {
+		if !fs.Changed(flagName) {
+			unchangedFlags = append(unchangedFlags, fmt.Sprintf("--%s", flagName))
+		}
+	}
+	nUnchangedFlags := len(unchangedFlags)
+	if nUnchangedFlags > 0 && nUnchangedFlags != len(requiredTogetherFlags) {
```

Comment:
Below condition is more aligned to non-or-all checking.
```suggestion
	if nUnchangedFlags != 0 && nUnchangedFlags != len(requiredTogetherFlags) {
```